### DISCUSSION TITLE
How to use rails_admin.dom_ready?

### DISCUSSION THREAD

**Discussion by gabrieletassoni** [railsadminteam/rails_admin#3600] [Q&A] [answered]
I cannot understand how to use that event, I mean, if I put this code in `app\assets\javascripts\rails_admin\custom\ui.js` this code:

```javascript
console.log("Ciao");

$(document).on('rails_admin.dom_ready', function () {
    console.log("Rails Admin DOM ready")
});
```

The *Ciao* word appears in console, but the phrase "Rails Admin DOM ready" never, even if I change page, I though the first time wasn't got fired because it was already fired when the event handler was registered, but I expect that rails_admin.dom_ready gets fired every time a new page is loaded.

What am I missing? which is the way of detecting page changes on rails admin javascript put into ui.js?

Thank you very much.

**Comment by gabrieletassoni** [comment]
Ended up on using turbo events to understand when a page visit has completed rendering and loading the page:

```javascript
let currentURL = window.location.href;

$(document).on('turbo:load', function (event) {
    currentURL = event.originalEvent.detail.url;
    console.log("Page loaded via turbo", currentURL);
});
```

I still cannot understand the use case for  `rails_admin.dom_ready`, but I think, for my needs, I'm ok with turbo events.

**Comment by mshibuya** [comment, marked as answer]
This is unexpected. And I found that the cause is the custom event name RailsAdmin use.
Currently it's `rails_admin.dom_ready` (separated by a period), unlike others which uses a colon (e.g. `turbo:load` for Turbo). And
 jQuery interprets a period-separated name as a [namespaced event](https://api.jquery.com/on/#event-names). So jQuery's `on` will listen to events with the type of `rails_admin` and the namespace `dom_ready`, not the event type `rails_admin.dom_ready`. Hence `$(document).on('rails_admin.dom_ready' ...)` doesn't get invoked.

I consider this as a regression and will attempt to fix. For now you can work around this by using `addEventListener` instead of jQuery's `on`.

```
document.addEventListener("rails_admin.dom_ready", function () {
    console.log("Rails Admin DOM ready")
});
```

**Comment by gabrieletassoni** [comment]
Thank you very much
